docs/composer.md

```diff
@@ -0,0 +1,86 @@
+---
+title: Composer
+tags:
+  - composer
+  - php
+  - dependency
+  - dependencies
+  - testing
+  - phpunit
+  - phpcs
+  - patch
+---
+# Composer
+
+[Composer][] is a tool for dependency management in PHP. It allows you to
+declare the libraries your project depends on and it will manage
+(install/update) them for you.
+
+## Repositories
+
+This project's `composer.json` file is configured to use packages.drupal.org and
+asset-packagist.org to pull in dependencies through [Composer][]. Drupal
+extensions can be installed via [Composer][] using the `drupal/*` package name.
+Likewise, npm and bower packages can be installed from asset-packagist.org using
+the `bower-asset/*` or `npm-asset/*` package name.
+
+## Patches
+
+You can patch composer dependencies with a local or remote patch file by
+defining patches in `composer.json`. Patches are defined as follows:
+
+```json
+{
+    "extra": {
+        "patches": {
+            "vendor/package": {
+                "Description of patch": "https://domain.tld/files/file.patch"
+            }
+        }
+    }
+}
+```
+
+## Preserve paths
+
+Drupal core files by default are ignored from version control, but there may be
+a need to customize a file like `.htaccess`. If you add changes to a file like
+`.htaccess` you will need to add it to the `preserve-paths` directive in
+`composer.json` to prevent [Composer][] from overwriting your changes. Preserved
+paths are defined as follows:
+
+```json
+{
+    "extra": {
+        "preserve-paths": [
+            "web/.htaccess"
+        ]
+    }
+}
+```
+
+## PHPCS
```

Comment:
I think this would be better suited as a section called "Custom Scripts" or something. And then listing PHPCS and PHPUnit custom scripts below it with some description.

The PHPCS and PHPUnit headings just seem out of place to me in a Composer document.

Might be nit picking too much here, though. Up to you!